--- correctOutput/for_simple.s3	2015-01-14 09:24:38.000000000 +0100
+++ testOutput/for_simple.s3	2015-01-28 14:04:10.831997138 +0100
@@ -3,6 +3,7 @@
   (nil)
   FUNCTION()()
    TYPE(VOID)()
+    (nil)
    VARIABLE()("main")()
    (nil)
    STATEMENT_LIST()()
@@ -10,6 +11,7 @@
      STATEMENT()()
       DECLARATION_STATEMENT()()
        TYPE(INT)()
+        (nil)
        VARIABLE()("i")()
     STATEMENT()()
      FOR_STATEMENT()()
