--- correctOutput/function_simple.s3	2015-01-14 09:24:38.000000000 +0100
+++ testOutput/function_simple.s3	2015-01-28 14:04:11.523997118 +0100
@@ -4,6 +4,7 @@
    (nil)
    FUNCTION()()
     TYPE(VOID)()
+     (nil)
     VARIABLE()("main")()
     (nil)
     STATEMENT_LIST()()
@@ -19,6 +20,7 @@
        (nil)
   FUNCTION()()
    TYPE(VOID)()
+    (nil)
    VARIABLE()("function")()
    (nil)
    STATEMENT_LIST()()
