--- correctOutput/while_hard.s3	2015-01-14 09:24:38.000000000 +0100
+++ testOutput/while_hard.s3	2015-01-28 14:04:14.351997034 +0100
@@ -3,6 +3,7 @@
   (nil)
   FUNCTION()()
    TYPE(VOID)()
+    (nil)
    VARIABLE()("main")()
    (nil)
    STATEMENT_LIST()()
@@ -11,6 +12,7 @@
       STATEMENT()()
        DECLARATION_STATEMENT()()
         TYPE(INT)()
+         (nil)
         VARIABLE()("a")()
      STATEMENT()()
       ASSIGNMENT_STATEMENT()()
@@ -33,6 +35,7 @@
           STATEMENT()()
            DECLARATION_STATEMENT()()
             TYPE(INT)()
+             (nil)
             VARIABLE()("b")()
          STATEMENT()()
           ASSIGNMENT_STATEMENT()()
@@ -55,6 +58,7 @@
               STATEMENT()()
                DECLARATION_STATEMENT()()
                 TYPE(INT)()
+                 (nil)
                 VARIABLE()("c")()
              STATEMENT()()
               ASSIGNMENT_STATEMENT()()
