--- correctOutput/function_medium.s10	2015-02-24 16:27:42.000000000 +0100
+++ testOutput/function_medium.s10	2015-02-25 20:32:47.087975029 +0100
@@ -3,9 +3,21 @@
 Type checking assignment
 Type checking expression FUNC_CALL
 Type checking expression +
+Type error at:
+EXPRESSION (FUNC_CALL)
+Type error at:
+EXPRESSION (FUNC_CALL)
+Type error at:
+EXPRESSION (FUNC_CALL)
 Type checking assignment
 Type checking expression FUNC_CALL
 Type checking expression &&
+Type error at:
+EXPRESSION (FUNC_CALL)
+Type error at:
+EXPRESSION (FUNC_CALL)
+Type error at:
+EXPRESSION (FUNC_CALL)
 Type checking expression &&
 Type checking expression ||
 Type checking expression ||
