--- correctOutput/matrix_multiply.s10	2015-02-24 16:27:42.000000000 +0100
+++ testOutput/matrix_multiply.s10	2015-02-25 20:32:47.227975030 +0100
@@ -1,7 +1,11 @@
 Type checking assignment
 Type checking expression NEW
+Type error at:
+ASSIGNMENT_STATEMENT
 Type checking assignment
 Type checking expression NEW
+Type error at:
+ASSIGNMENT_STATEMENT
 Type checking assignment
 Type checking assignment
 Type checking assignment
@@ -9,19 +13,27 @@
 Type checking expression ARRAY_INDEX
 Type checking expression +
 Type checking expression *
+Type error at:
+ASSIGNMENT_STATEMENT
 Type checking assignment
 Type checking expression ARRAY_INDEX
 Type checking expression ARRAY_INDEX
 Type checking expression -
 Type checking expression +
 Type checking expression *
+Type error at:
+ASSIGNMENT_STATEMENT
 Type checking assignment
 Type checking expression NEW
+Type error at:
+ASSIGNMENT_STATEMENT
 Type checking assignment
 Type checking assignment
 Type checking assignment
 Type checking expression ARRAY_INDEX
 Type checking expression ARRAY_INDEX
+Type error at:
+ASSIGNMENT_STATEMENT
 Type checking assignment
 Type checking assignment
 Type checking expression ARRAY_INDEX
@@ -34,6 +46,12 @@
 Type checking expression ARRAY_INDEX
 Type checking expression ARRAY_INDEX
 Type checking expression ARRAY_INDEX
+Type error at:
+EXPRESSION (*)
+Type error at:
+EXPRESSION (+)
+Type error at:
+ASSIGNMENT_STATEMENT
 Type checking expression ARRAY_INDEX
 Type checking expression ARRAY_INDEX
 Type checking expression ARRAY_INDEX
